src/rgw/rgw_sal_motr.cc

```diff
@@ -2737,18 +2731,45 @@ int MotrAtomicWriter::complete(size_t accounted_size, const std::string& etag,
   string tenant_bkt_name = get_bucket_name(obj.get_bucket()->get_tenant(), obj.get_bucket()->get_name());
   // Insert an entry into bucket index.
   string bucket_index_iname = "motr.rgw.bucket.index." + tenant_bkt_name;
+
+  if( !info.versioning_enabled() )
+  {
+	std::string obj_type = "simple object";
+  	rgw_bucket_dir_entry old_ent;
+  	bufferlist old_check_bl;	
+	// Before updating bucket index with entry for new object, check if
+  	// old object exists. Perform M0_IC_GET operation on bucket index.
+  	rc = store->do_idx_op_by_name(bucket_index_iname, M0_IC_GET,
+                                obj.get_name(), old_check_bl);
+  	if (rc == 0 && old_check_bl.length() > 0) {
+    		auto ent_iter = old_check_bl.cbegin();
+    		old_ent.decode(ent_iter);
+    		std::unique_ptr<rgw::sal::Object> old_obj = obj.get_bucket()->get_object(rgw_obj_key(obj.get_name()));
+    		rgw::sal::MotrObject *old_mobj = static_cast<rgw::sal::MotrObject *>(old_obj.get());
+    		if (old_ent.meta.category == RGWObjCategory::MultiMeta) {
+      			obj_type = "multipart object";
+      			old_mobj->set_category(RGWObjCategory::MultiMeta);
+    		}
+    	ldpp_dout(dpp, 20) <<__func__<< ": Old " << obj_type << " exists" << dendl;
+    	// Delete old object
+	// Initialize version of the previous object so that it can delete a specific version,rather than delete the latest object.
+	// TODO: This will be taken care during version implementation.
+    	rc = old_mobj->delete_object(dpp, NULL, y, true);
+	if (rc == 0) {
+      		ldpp_dout(dpp, 20) <<__func__<< ": Old " << obj_type << " ["
+          	<< old_mobj->get_name() <<  "] deleted succesfully" << dendl;
+    	} else {
+      		ldpp_dout(dpp, 0) << __func__<<": Failed to delete old " << obj_type << " ["
+          	<< old_mobj->get_name() <<  "]. Error = " << rc << dendl;
+                // TODO: This will be handled during GC
+    	}
+  	}
+  }
   rc = store->do_idx_op_by_name(bucket_index_iname,
```

Comment:
As discussed we wanted to do this op first before deleting old object  .. rt ?

Reply:
We can not achieve this with the current delete object function design
That will require objects to be created with version id so that we can delete the specific version. Which will again will be taken care by GC in case of any failure.

For now we will try to delete the latest object, if there is any error, we will skip it and proceed with writing the new object's entry

Created a ticket for GC - https://jts.seagate.com/browse/CORTX-31829

Also created one more for versioning team.
https://jts.seagate.com/browse/CORTX-31830